modelforge/train/training.py

```diff
@@ -183,15 +189,25 @@ def _predict_dipole_moment(
         self, model_predictions: Dict[str, torch.Tensor], batch: BatchData
     ) -> torch.Tensor:
 
-        per_atom_charge = model_predictions["per_atom_charge"]
+        per_atom_charge = model_predictions["per_atom_charge"]  # Shape: [num_atoms]
+        positions = batch.nnp_input.positions  # Shape: [num_atoms, 3]
+        per_atom_charge = per_atom_charge.unsqueeze(-1)  # Shape: [num_atoms, 1]
+        per_atom_dipole_contrib = per_atom_charge * positions  # Shape: [num_atoms, 3]
+
+        indices = batch.nnp_input.atomic_subsystem_indices.long()  # Shape: [num_atoms]
+        indices = indices.unsqueeze(-1).expand(-1, 3)  # Shape: [num_atoms, 3]
+
```

Comment:
Listing all the shapes is very useful!

Reply:
I hope that we can do this by using custom tensor classes in the future. this would help a lot and would implicitly document shapes (especially (nr_atoms, 1), (nr_atom, 3), (n_pairs,1), (n_pairs, n_features) are common enough that this might make sense